src/briefcase/commands/new.py

```diff
@@ -327,6 +411,43 @@ def build_app_context(self):
         # The module name can be completely derived from the app name.
         module_name = self.make_module_name(app_name)
 
+        default_source_dir = f"src/{module_name}"
```

Comment:
This workflow feels a little backwards. If a user has specified a `--target`, and that folder contains a `pyproject.toml` or `setup.cfg`, then we should be able to extract a candidate formal name from that metadata, and use that to generate a candidate appname. 

At the very least, there should be some acknowledgement in the prompt text that we're converting an existing app. The current process start with "Let's build a new Briefcase app", which doesn't suggest anything about migrating and existing app. In my initial testing, I wasn't sure if I'd activated the right branch, because the workflow was identical until I was asked for a source directory.

Reply:
Good point! So far we have changed the text descriptions to reflect that we are setting up an existing app is set up for Briefcase whenever the `--target` directory is provided and it exists. We could also try to extract the metadata if they have a pyproject.toml file that follows [PEP621](https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#declaring-project-metadata). What do you think?

Reply:
I'll likely tweak the specific text before landing this, but this definitely sounds like the right direction. In particular, we should be "hinting" as much as possible from as many sources as possible - if PEP621 metadata exists, we should be using it.